packages/back-end/src/entities/database.ts

```diff
@@ -0,0 +1,234 @@
+        { id: id }
+      );
+
+      const partners = result.records.map((record) => record.get("partner"));
+
+      return partners;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findChildren(person: number | Person): Promise<Person[]> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)<-[:FATHER|:MOTHER]-(child:Person) WHERE p.id = $id RETURN child",
+        { id: id }
+      );
+
+      const children = result.records.map((record) => record.get("child"));
+
+      return children;
+    }
+    finally {
+      await session.close();
+    }
+  }
+
+  async updatePerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person) WHERE p.id = $id SET p = $person",
+        { id: person.id, person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async createPerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "CREATE (p:Person $person)",
+        { person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async setFather(person: Person, father: Person): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person)-[r:FATHER]->(f:Person) WHERE p.id = $personId DELETE r",
+        { personId: person.properties.id }
+      );
+      await session.run(
+        "MATCH (p:Person), (f:Person) WHERE p.id = $personId AND f.id = $fatherId CREATE (p)-[:FATHER]->(f)",
+        { personId: person.properties.id, fatherId: father.properties.id }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async setMother(person: Person, mother: Person): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person)-[r:MOTHER]->(m:Person) WHERE p.id = $personId DELETE r",
+        { personId: person.properties.id }
+      );
+      await session.run(
+        "MATCH (p:Person), (m:Person) WHERE p.id = $personId AND m.id = $motherId CREATE (p)-[:MOTHER]->(m)",
+        { personId: person.properties.id, motherId: mother.properties.id }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async addPartner(person: Person, partner: Person): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person), (partner:Person) WHERE p.id = $personId AND partner.id = $partnerId CREATE (p)-[:PARTNER]->(partner)",
+        { personId: person.properties.id, partnerId: partner.properties.id }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async removePartner(person: Person, partner: Person): Promise<void> {
```

Comment:
add removeFather removeMother to enable corrections

Reply:
👍